src/Angor/Client/Pages/Signatures.razor

```diff
@@ -160,6 +174,33 @@
         }
     }
 
+    private async Task RefreshSignatures()
+    {
+        if (isCooldown)
+            return;
+        isLoading = true;
+        StateHasChanged();
+        isCooldown = true;
+        await FetchPendingSignatures(FounderProject);
+
+
+        if (!await EnsurePassword())
```

Comment:
What you should do is a method like this above

```
 private async Task RefreshSignaturesInternal()
    {
        
            if (passwordComponent.HasPassword())
            {
                await RefreshSignatures();
            }
            else
            {
                passwordComponent.ShowPassword(RefreshSignatures);
            }
        
    }

```

Reply:
I thought it might be code duplication, but on second thought, it significantly improves readability.
theres no need to be overly concerned with saving lines of code here. changed.